From f386938847842b30ad9f7df2e9d58a0d1338849a Mon Sep 17 00:00:00 2001
From: Tao Zeng <tao.zeng@amlogic.com>
Date: Mon, 17 Aug 2015 13:38:12 +0800
Subject: [PATCH 5775/5965] PD#110962: cec: fix kernel panic for auto-reboot
 test

Add a condition check to avoid pointer index out of buffer size.
This will cause kernel panic probabilistic when doing auto-reboot
test.

Change-Id: I96e184e1db3349424ce90728cee2a8405151154c
---
 drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c
index 8d411f9924cd..baef2791bf5e 100755
--- a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c
+++ b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c
@@ -2052,7 +2052,7 @@ void cec_usrcmd_set_config(const char * buf, size_t count)
             param[j] = simple_strtoul(&buf[i], NULL, 16);
             j ++;
         }
-        while ( buf[i] != ' ' )
+        while (i < count && buf[i] != ' ')
             i ++;
     }
     value = aml_read_reg32(P_AO_DEBUG_REG0);
-- 
2.19.0

